### PR TITLE
Debug logging for end_of_epoch

### PR DESCRIPTION
**Pull request by zdevito**
In larger runs some dataloaders appear to end early. Adding some debug logging when they reach an end_of_epoch state to try to spot the difference in these instances vs others.

### CODE REVIEW

File: metaseq/cli/train.py [hunk 1]
```diff
@@ -274,6 +274,16 @@ def train(
                 metrics.reset_meters("train_inner")
 
         end_of_epoch = not itr.has_next()
+        if end_of_epoch:
+            end_of_epoch = distributed_utils.get_global_rank()
```
Comment (stephenroller) [inline metaseq/cli/train.py:278]: I think you mean
```suggestion
            grank = distributed_utils.get_global_rank()
```

File: metaseq/cli/train.py [hunk 2]
```diff
@@ -274,6 +274,16 @@ def train(
                 metrics.reset_meters("train_inner")
 
         end_of_epoch = not itr.has_next()
+        if end_of_epoch:
```
Comment (stephenroller) [inline metaseq/cli/train.py:277]: can we leave a comment pointing to this PR or the task and saying we want to delete this after we resolve the issue?

Comment (zdevito) [reply metaseq/cli/train.py:277]: Should we merge it, or do you just want to cherry-pick it into a job run? I am fine either way

**Review by suchenzang — APPROVED** [review]
Nit on lint - @stephenroller are you pulling this into our job branch?

**Comment by stephenroller** [comment]
Pulled into bigbig manually.

**Pull request merged**